flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/FinishedOperatorChain.java

```diff
@@ -62,8 +63,9 @@ public WatermarkGaugeExposingOutput<StreamRecord<OUT>> getMainOperatorOutput() {
     }
 
     @Override
-    public void dispatchOperatorEvent(OperatorID operator, SerializedValue<OperatorEvent> event) {
-        throw new UnsupportedOperationException();
+    public void dispatchOperatorEvent(OperatorID operator, SerializedValue<OperatorEvent> event)
```

Comment:
It seems weird that a `FinishedOperatorChain` now needs to support event dispatch. Conceptually a finished operator chain (or task) should not have to send/receive anything.

Prior to this PR, does `FinishedOperatorChain` need to send/receive any event? If no, is there anyway to keep the same behavior?

If we have to change the behavior in order to support `CloseGatewayEvent/AcknowledgeCloseGatewayEvent`, can you explain the main difference between these two messages and all existing control flow messages that make this change necessary?

Reply:
After making all other changes that possibly fixed some bugs, I found that the `FinishedOperatorChain:: dispatchOperatorEvent` method would not be invoked anymore. Thus I'll make this method throw `UnsupportedOperationException` as before.